var/spack/repos/builtin/packages/berkeley-db/package.py

```diff
@@ -32,3 +34,23 @@ class BerkeleyDb(AutotoolsPackage):
 
     def configure_args(self):
         return ['--disable-static', '--enable-cxx', '--enable-stl']
+
+    def _test_check_versions(self):
+        """Perform version checks on selected installed package binaries."""
+        exes = [
+            'db_checkpoint', 'db_deadlock', 'db_dump', 'db_load',
+            'db_printlog', 'db_stat', 'db_upgrade', 'db_verify'
+        ]
+        for exe in exes:
+            reason = 'test version of {0} is {1}'.format(exe,
+                                                         self.spec.version)
+            self.run_test(exe, ['-V'], [self.spec.version.string], None,
+                          installed=True, purpose=reason, skip_missing=True)
+
+    def test(self):
+        """Perform smoke tests on the installed package."""
+        tty.debug('Expected results currently based on simple {0} builds'
+                  .format(self.name))
```

Comment:
Does this really add anything useful for the user

Reply:
In this case at this time, no.  (Should've put my "standard" version range debug message instead. ;))